### PR TITLE
Extra scripts - gather_statistics

### PR DESCRIPTION
**Pull request by pdellaert**
Another example i created, this time it will gather stats depending on what is requested.

### CODE REVIEW

File: gather_statistics.py [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+        nc.start()
+
+    except Exception, e:
+        logger.error('Could not connect to Nuage host %s with user %s and specified password' % (nuage_host, nuage_username))
+        logger.critical('Caught exception: %s' % str(e))
+        return 1
+
+    # Getting entities & defining output fields
+    output_type = None
+    entities = []
```
Comment (primalmotion) [inline gather_statistics.py]: You could simplify all that block by using `fetcher_for_rest_name()`

For instance:

```python
entities = []
search_query = 'name == %s' % entity_name if entity_name else None
entities = entities + nc.user.fetcher_for_rest_name(entity_type.lower()).get(filter=search_query)
output_type = entity_name.capitalize()
```

Comment (pdellaert) [inline gather_statistics.py]: Thanks, didn't know about that trick ;). Let me have a look at it and I'll update the pull request.

Comment (primalmotion) [inline gather_statistics.py]: You should take a look at the bambou source code, there's a lot of little things like that ;)

Comment (pdellaert) [inline gather_statistics.py]: I'll try and spent some time on that :). Cheers!

**Comment by primalmotion** [comment]
Great, Thanks!

**Pull request merged**